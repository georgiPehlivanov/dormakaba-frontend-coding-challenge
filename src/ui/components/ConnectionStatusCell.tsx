import { ConnectionStatus } from '@/models/ConnectionStatus';
import { Typography } from '@mui/material';

interface ConnectionStatusCellProps {
  connectionStatus: ConnectionStatus;
}

export function ConnectionStatusCell({
  connectionStatus,
}: ConnectionStatusCellProps) {
  const isOnline = connectionStatus === ConnectionStatus.Online;

  return (
    <Typography
      component="span"
      color={isOnline ? 'success.main' : 'error.main'}
    >
      {connectionStatus}
    </Typography>
  );
}
