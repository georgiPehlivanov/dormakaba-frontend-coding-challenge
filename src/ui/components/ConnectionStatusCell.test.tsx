import { render } from '@testing-library/react';
import { ConnectionStatusCell } from './ConnectionStatusCell';
import { ConnectionStatus } from '@/models/ConnectionStatus';

const connectionStatus: ConnectionStatus = ConnectionStatus.Online;

describe('ConnectionStatusCell', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ConnectionStatusCell connectionStatus={connectionStatus} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
