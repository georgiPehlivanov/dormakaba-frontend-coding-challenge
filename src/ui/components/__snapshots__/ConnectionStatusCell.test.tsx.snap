// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectionStatusCell should render correctly 1`] = `
<span
  class="MuiTypography-root MuiTypography-body1 css-wl91uy-MuiTypography-root"
>
  online
</span>
`;
